code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm

```diff
@@ -120,6 +121,527 @@ datum/reagent/toxin/slimejelly/on_mob_life(var/mob/living/M as mob)
+datum/reagent/toxin/smokep/on_fart(var/mob/living/M as mob)
+	M.visible_message("<span class='danger'>[M] farts out a noxious cloud!</span>")
+	create_smoke(src, 10)
+
+datum/reagent/toxin/fart//for wu eric
+	name = "Fartium"
+	id = "fart"
+	description = "Hmm I wonder?"
+	color = "#10AC53"
+datum/reagent/toxin/fart/on_mob_life(var/mob/living/M as mob)
+	if(prob(20))
+		M.emote("fart")//yup thats it
+datum/reagent/toxin/sboom
+	name = "Nitrogenated isopropyl alcohol"
+	id = "sboom"
+	description = "Hmm , needs more nitrogen!"
+	color = "#13BC5E"
+datum/reagent/toxin/superboom
+	name = "N-amino azidotetrazole"
+	id = "superboom"
+	description = "If this stuff is even slightly warmed it will make a mess."
+	color = "#13BC5E"
+datum/reagent/toxin/superboom/on_mob_life(var/mob/living/M as mob)
+	if(M.bodytemperature > 285)//you CAN avoid it , just it's difficult
+		var/location = get_turf(holder.my_atom)
+		explosion(location,volume/20,volume/15,volume/10)
+		holder.clear_reagents()
+		return
+
+
+datum/reagent/toxin/superboom/reaction_mob(var/mob/living/M, var/method=TOUCH, var/volume)
+	if(M.bodytemperature > 285)
+		var/location = get_turf(holder.my_atom)
+		explosion(location,volume/20,volume/15,volume/10)
+		holder.clear_reagents()
+		return
+datum/reagent/toxin/superboom/on_update()
+	if(!holder)
+		return
+	if(holder.present_machines[1] >= 285)
+		var/location = get_turf(holder.my_atom)
+		explosion(location,volume/20,volume/15,volume/10)
+		holder.clear_reagents()
+		return
+
+datum/reagent/toxin/ethyl
+	name = "Ethyl alcohol"
+	id = "ethyl"
+	description = "Brilliant at burning"
+	color = "#A245B2"
+datum/reagent/toxin/carbonf
+	name = "Carbonic fluoride"
+	id = "carbonf"
+	description = "A fairly nasty chemical used to produce potent medicines"
+	color = "#A300B3"
+datum/reagent/toxin/carbonf/on_mob_life(var/mob/living/M as mob)
+	M.adjustToxLoss(1)
+datum/reagent/toxin/sparky
+	name = "Electrostatic substance"
+	id = "sparky"
+	description = "The lights glow nearby when this substance is around"
+	color = "#A300B3"
+
+datum/reagent/toxin/spark/on_mob_life(var/mob/living/M as mob)
+	if(prob(10))
+		M.adjustFireLoss(3)//extremely weak damage
+		var/datum/effect/effect/system/spark_spread/sparks = new /datum/effect/effect/system/spark_spread
+		sparks.set_up(1, 1, src)
+		sparks.start()
+	if(prob(5))
+		M.IgniteMob()
+		M.adjust_fire_stacks(3)
+		holder.remove_reagent(src.id, 5, safety = 1)
+datum/reagent/toxin/dizinc
+	name = "Diethyl zinc"
+	id = "dizinc"
+	description = "Definately dont spray this at the clown"
+	color = "#000067"
+datum/reagent/toxin/dizinc/on_mob_life(var/mob/living/M as mob)
+	M.adjust_fire_stacks(6)
+	M.IgniteMob()
+	M.bodytemperature += 15 //it's getting HOT HOT HOT dunununu
+	holder.remove_reagent(src.id, 5, safety = 1)
+	return
+datum/reagent/toxin/dizinc/reaction_mob(var/mob/living/M, var/method=TOUCH, var/volume)
+	M.adjust_fire_stacks(6)
+	M.IgniteMob()
+	holder.clear_reagents()
```

Comment:
why would I use any anti-toxin over this if I have a spray bottle of water on hand